dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalIterable.java

```diff
@@ -108,9 +139,26 @@ public static ZonedIntervalIterable forConnectorArguments(@Nonnull ConnectorArgu
           "At least one day of query logs should be exported; you specified: " + daysToExport);
 
     LOG.info(
-        "Log entries within the last {} days will be exported in increments of 1 hour.",
-        daysToExport);
-    return ZonedIntervalIterable.forTimeUnitsUntilNow(24 * daysToExport, ChronoUnit.HOURS);
+        "Log entries within the last {} days will be exported in increments of 1 {}.",
+        daysToExport,
+        timeUnit);
+
+    int multiplicationFactor;
+
+    switch (timeUnit) {
+      case DAYS:
+        multiplicationFactor = 1;
+        break;
+      case HOURS:
+        multiplicationFactor = 24;
+        break;
+      default:
+        throw new IllegalStateException(
```

Comment:
This switch and throw can be combined with the "if (!SUPPORTED_UNITS.contains(timeUnit))" check above.